api/v1alpha1/istio_types.go

```diff
@@ -28,6 +28,12 @@ const IstioKind = "Istio"
 // IstioSpec defines the desired state of Istio
 type IstioSpec struct {
 	Version string `json:"version,omitempty"`
+
+	// The built-in installation configuration profile to use. One of:
+	// default, demo, minimal, remote, empty, preview, ambient.
+	// When this field is left empty, the 'default' profile will be used.
+	Profile string `json:"profile,omitempty"`
```

Comment:
What about the openshift profile? Keep in mind that this is an upstream project, we can't default to openshift. Or at least we shouldn't, IMO.

Reply:
We don't default to OpenShift, but we do override some values [here](https://github.com/Maistra/istio-operator/blob/fff1734ab6053c8a43b8624b386108d19dc3a14e/pkg/strategy/maistra_30_strategy.go#L20-L47). The plan is to eventually remove those overrides.

Reply:
this is wrong, no? We shouldn't be overriding values such as cni.enabled I think.

Reply:
I agree. But this is how this was done initially and changing it isn't part of this PR.